lib/simp/rpm.rb

```diff
@@ -27,15 +28,58 @@ def self.sh(args)
     #           encountered!
     # [full_version] The full version of the package: [version]-[release]
     # [name] The full name of the package: [basename]-[full_version]
+    # [packages] An Array of package names belonging to the spec file or RPM
     def initialize(rpm_source)
       info = Simp::RPM.get_info(rpm_source)
+
       @basename = info[:name]
       @version = info[:version]
       @release = info[:release]
       @full_version = info[:full_version]
       @name = "#{@basename}-#{@full_version}"
-      @sources = Array.new
+      @arch = info[:arch]
+      @package_name = "#{@name}.#{@arch}.rpm"
       @verbose = false
+      @packages = info[:packages]
+      @signature = info[:signature]
+    end
+
+    # Returns whether or not the current RPM is newer than the passed RPM
+    def newer?(rpm)
+      return Simp::RPM.newer(@package_name, rpm) == @package_name
+    end
+
+    # Returns whether or not the first RPM is newer than the second RPM
+    def self.newer?(rpm1, rpm2)
+      return Simp::RPM.newer(rpm1, rpm2) == rpm1
+    end
+
+    # Returns the name of the newer RPM
+    #
+    # Will simply return the second RPM name if the two are equal
+    def self.newer(rpm1, rpm2)
+      if rpm1.nil? || rpm1.empty?
+        fail('You must pass at least one RPM')
+      end
+
+      return rpm1 if rpm2.nil? || rpm2.empty?
+
+      unless rpm1.match(%r(\.rpm$)) && rpm2.match(%r(\.rpm$))
+        fail("You must pass valid RPM names! Got: '#{rpm1}' and '#{rpm2}'")
+      end
+
+      begin
+        ver1 = Gem::Version.new(File.basename(rpm1, '.rpm').split('-')[-1..-1].join('-'))
+        ver2 = Gem::Version.new(File.basename(rpm2, '.rpm').split('-')[-1..-1].join('-'))
+
+        if ver1 > ver2
+          return ver1
+        else
+          return ver2
+        end
+      rescue ArgumentError, NoMethodError
+        fail("Could not compare RPMs '#{rpm1}' and '#{rpm2}'")
```

Comment:
This fails for simp RPM when I called pkg:check_published target.  Output is:
Could not compare RPMs 'simp-6.1.0-RC1.el7.noarch.rpm' and 'simp-6.0.0-1.el7.noarch.rpm'.
Had to replace the fail with a $stderr.puts in order to see what pkg:check_published did with modules.